### PR TITLE
add 'questions' for save to case attribs

### PR DESCRIPTION
**Pull request by snopoke**
This adds `question` elements returned from `xform.get_questions` for `@case_id`, `@user_id` and `@date_modified` or save to case blocks.

This mostly impacts data exports which previously couldn't export these fields.

### CODE REVIEW

File: corehq/apps/reports/tests/readable_forms/top_level_refless_group.result.yaml [hunk 1]
```diff
@@ -764,14 +765,12 @@
...
 - {calculate: null, comment: null, group: null, label: note/discussed_appointments,
   relevant: null, repeat: null, required: null, response: 'no', tag: null, type: null,
   value: /progress_note/note/discussed_appointments}
 - {calculate: null, comment: null, group: null, label: note/discussed_followup, relevant: null,
-  repeat: null, required: null, response: 'no', tag: null, type: null, value:
-  /progress_note/note/discussed_followup}
+  repeat: null, required: null, response: 'no', tag: null, type: null, value: /progress_note/note/discussed_followup}
 - {calculate: null, comment: null, group: null, label: note/discussed_goals, relevant: null,
   repeat: null, required: null, response: 'no', tag: null, type: null, value: /progress_note/note/discussed_goals}
 - {calculate: null, comment: null, group: null, label: note/discussed_qsp, relevant: null,
```
Comment (snopoke) [inline corehq/apps/reports/tests/readable_forms/top_level_refless_group.result.yaml]: @dannyroberts all these changes are noop changes due to ordering of dict keys. I didn't have time to dig into how to force the ordering. Do you happen to know?

This is generated here: https://github.com/dimagi/commcare-hq/blob/3e88dbe4554316981e7f9a43fde34aa8870702c9/corehq/apps/reports/tests/test_readable_formdata.py#L268

Comment (dannyroberts) [reply corehq/apps/reports/tests/readable_forms/top_level_refless_group.result.yaml]: I think that the line you referenced _does_ force ordering.

```
>>> print yaml.safe_dump([{'e': 5, 'f': 6, 'g': 7, 'a': 1, 'b': 2, 'c': 3, 'd': 4, 'A': 123, 'B': 123}])
- {A: 123, B: 123, a: 1, b: 2, c: 3, d: 4, e: 5, f: 6, g: 7}
```
Could it be that the last person to edit these files generated them in a different way?

**Review by dannyroberts — APPROVED** [review]

File: corehq/apps/reports/tests/readable_forms/top_level_refless_group.result.yaml (hunk 1 above)
Comment (snopoke) [reply corehq/apps/reports/tests/readable_forms/top_level_refless_group.result.yaml]: Thanks, sorry I clearly didn't look closely enough.

Comment (dannyroberts) [reply corehq/apps/reports/tests/readable_forms/top_level_refless_group.result.yaml]: 👍 no worries

**Pull request merged**